### ISSUE TITLE
New test error with urllib3 1.25

### ISSUE THREAD

**Issue by opoplawski** [urllib3/urllib3#1575]
I'm looking at updating urllib3 to 1.25 in Fedora EPEL7.  I'm getting the following test errors:
```
platform linux -- Python 3.6.6, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /usr/bin/python3.6
=================================== FAILURES ===================================
 TestConnectionPool.test_same_host_custom_protocol[%2Fvar%2Frun%2Fdocker.sock-http+unix://%2Fvar%2Frun%2Fdocker.sock] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee3ba59e8>
a = '%2Fvar%2Frun%2Fdocker.sock', b = 'http+unix://%2Fvar%2Frun%2Fdocker.sock'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba5b00>>('http+unix://%2Fvar%2Frun%2Fdocker.sock')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba5b00>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba5b00>.is_same_host

test_connectionpool.py:170: AssertionError
 TestConnectionPool.test_same_host_custom_protocol[%2Fvar%2Frun%2Fdocker.sock-http+unix://%2Fvar%2Frun%2Fdocker.sock/] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee3af46a0>
a = '%2Fvar%2Frun%2Fdocker.sock', b = 'http+unix://%2Fvar%2Frun%2Fdocker.sock/'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3aea4a8>>('http+unix://%2Fvar%2Frun%2Fdocker.sock/')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3aea4a8>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3aea4a8>.is_same_host

test_connectionpool.py:170: AssertionError
 TestConnectionPool.test_same_host_custom_protocol[%2Fvar%2Frun%2Fdocker.sock-http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee5029e48>
a = '%2Fvar%2Frun%2Fdocker.sock'
b = 'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba53c8>>('http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba53c8>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3ba53c8>.is_same_host

test_connectionpool.py:170: AssertionError
 TestConnectionPool.test_same_host_custom_protocol[%2Ftmp%2FTEST.sock-http+unix://%2Ftmp%2FTEST.sock] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee68b2f98>
a = '%2Ftmp%2FTEST.sock', b = 'http+unix://%2Ftmp%2FTEST.sock'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b9ab00>>('http+unix://%2Ftmp%2FTEST.sock')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b9ab00>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b9ab00>.is_same_host

test_connectionpool.py:170: AssertionError
 TestConnectionPool.test_same_host_custom_protocol[%2Ftmp%2FTEST.sock-http+unix://%2Ftmp%2FTEST.sock/] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee50129e8>
a = '%2Ftmp%2FTEST.sock', b = 'http+unix://%2Ftmp%2FTEST.sock/'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b70518>>('http+unix://%2Ftmp%2FTEST.sock/')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b70518>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b70518>.is_same_host

test_connectionpool.py:170: AssertionError
 TestConnectionPool.test_same_host_custom_protocol[%2Ftmp%2FTEST.sock-http+unix://%2Ftmp%2FTEST.sock/abracadabra] 

self = <test.test_connectionpool.TestConnectionPool object at 0x7f8ee5029da0>
a = '%2Ftmp%2FTEST.sock', b = 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'

    @pytest.mark.parametrize('a, b', [
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/'),
        ('%2Fvar%2Frun%2Fdocker.sock',
         'http+unix://%2Fvar%2Frun%2Fdocker.sock/abracadabra'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/'),
        ('%2Ftmp%2FTEST.sock', 'http+unix://%2Ftmp%2FTEST.sock/abracadabra'),
    ])
    def test_same_host_custom_protocol(self, a, b):
        with HTTPUnixConnectionPool(a) as c:
>           assert c.is_same_host(b)
E           assert <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b99d30>>('http+unix://%2Ftmp%2FTEST.sock/abracadabra')
E            +  where <bound method HTTPConnectionPool.is_same_host of <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b99d30>> = <test.test_connectionpool.HTTPUnixConnectionPool object at 0x7f8ee3b99d30>.is_same_host

test_connectionpool.py:170: AssertionError
```
Not sure if these are new tests with 1.25, or just an issue with older support libraries, or ?

**Comment by opoplawski** [comment]
In particular we are stuck with an older pytest.

**Comment by sethmlarson** [comment]
I don't think it's pytest that's causing that failure and I'm unable to reproduce it locally. I'm not familiar with how Fedora EPEL packages urllib3, do you debundle libraries?

**Comment by sethmlarson** [comment]
Found an issue in rfc3986 and how we bundle. Will resolve those issues.

**Comment by sethmlarson** [comment]
This issue is fixed in 1.25.1.
